deployment/docker/Dockerfile

```diff
@@ -1,53 +1,58 @@
-#--------- Generic stuff all our Dockerfiles should start with so we get caching ------------
-FROM python:2.7.18-buster
-MAINTAINER Admire Nyakudya<admire@kartoza.com>
+# Use the official Python image with version 3.11 as the base image
+FROM python:3.11-buster
 
+# Set the maintainer label
+LABEL maintainer="Admire Nyakudya <admire@kartoza.com>"
 
 # Install system dependencies
 RUN apt-get update && \
-    apt-get install -y libgdal-dev libcurl4-gnutls-dev librtmp-dev libssl-dev  binutils libproj-dev gdal-bin &&  \
-    pip install uwsgi setuptools;pip uninstall -y wheel
+    apt-get install -y libgdal-dev libcurl4-gnutls-dev librtmp-dev libssl-dev binutils libproj-dev gdal-bin && \
+    pip install uwsgi setuptools && pip uninstall -y wheel
 
+# Install Node.js and npm
+RUN curl -sL https://deb.nodesource.com/setup_18.x -o nodesource_setup.sh && \
+    bash nodesource_setup.sh && \
+    apt-get install nodejs -y
 
-RUN curl -sL https://deb.nodesource.com/setup_18.x -o nodesource_setup.sh
-RUN bash nodesource_setup.sh
-RUN apt-get install nodejs -y
-
+# Copy your Django project to /home/web/django_project
 ADD ./django_project /home/web/django_project
+
+# Copy requirements.txt to the root directory
 ADD ./deployment/docker/requirements.txt /REQUIREMENTS.txt
+
+# Install Python dependencies from requirements.txt
 RUN pip install --no-deps -r /REQUIREMENTS.txt
 
-ADD ./deployment/docker/libgeos.py /usr/local/lib/python2.7/site-packages/django/contrib/gis/geos/libgeos.py
-ADD ./deployment/docker/trans_real.py /usr/local/lib/python2.7/site-packages/django/utils/translation/trans_real.py
-#RUN sed -i 's/geos_version().decode()/geos_version().decode().split(\x27 \x27)/g' /usr/local/lib/python2.7/site-packages/django/contrib/gis/geos/libgeos.py
-#USER www-data
-# ADD ./django_project /home/web/django_project
+# Copy necessary files to the appropriate locations
+ADD ./deployment/docker/libgeos.py /usr/local/lib/python3.11/site-packages/django/contrib/gis/geos/libgeos.py
+ADD ./deployment/docker/trans_real.py /usr/local/lib/python3.11/site-packages/django/utils/translation/trans_real.py
 
 # Set the working directory inside the container
 WORKDIR /app
 
-# Copy the package.json and package-lock.json files to the container
+# Copy package.json and package-lock.json to the container
 COPY ./django_project/minisass_frontend/package.json ./django_project/minisass_frontend/package-lock.json /app/
 
-# Install project dependencies
+# Install frontend dependencies
 RUN npm install
 
-
 # Copy the rest of the frontend source code to the container
 COPY ./django_project/minisass_frontend /app
 
 # Build the frontend
 RUN npm run build
 
-
 WORKDIR /home/web/django_project
-# RUN python manage.py collectstatic --noinput
+
+# Add uwsgi configuration and run script
 ADD ./deployment/docker/uwsgi.conf /uwsgi.conf
 ADD ./deployment/docker/run.sh /run.sh
-RUN chmod 0755 /run.sh
 
+# Make the run.sh script executable
+RUN chmod 0755 /run.sh
```

Comment:
The `run.sh` can also be simplified. Currently, it has the option to run development settings and production. Maybe we can have the entrypoint to be `https://github.com/kartoza/miniSASS/blob/main/deployment/docker/run.sh#L8`